src/gluonts/ext/hierarchicalforecast/_predictor.py

```diff
@@ -0,0 +1,309 @@
+        Options are `normality`, `bootstrap`, `permbu`.
+    quantile_levels
+        Optional list of quantile levels that we want predictions for.
+        Note: this is only supported by specific types of models, such as
+        ``AutoARIMA``. By default this is ``None``, giving only the mean
+        prediction.
+    n_jobs
+        number of jobs used in the parallel processing, use -1 for all cores.
+    model_params
+        dictionary with inputs parameters for base_model,
+        i.e. {"season_length": 2}
+    reconciler_params
+        dictionary with input parameters for reconciler,
+        i.e. {"method": "average_proportions"}
+    """
+
+    @validated()
+    def __init__(
+        self,
+        prediction_length: int,
+        base_model: Any,
+        reconciler: Any,
+        S: Union[List[List[int]], np.ndarray],
+        tags: Dict[str, Union[List, np.ndarray]] = {},
+        ts_names: Optional[List[Union[str, Any]]] = None,
+        intervals_method: str = "normality",
+        quantile_levels: Optional[List[float]] = None,
+        n_jobs: int = 1,
+        model_params: dict = {},
+        reconciler_params: dict = {},
+    ) -> None:
+        super().__init__(prediction_length=prediction_length)
+
+        assert intervals_method in ["normality", "bootstrap", "permbu"]
+
+        if tags and ts_names:
+            assert set([x for values in tags.values() for x in values]) == set(
+                ts_names
+            ), "tags and ts_names must have the same set of ts names"
+
+        self.models = [base_model(**model_params)]
+        self.hrec = HierarchicalReconciliation(
+            reconcilers=[reconciler(**reconciler_params)]
+        )
+        self.ts_names = (
+            ts_names
+            if ts_names is not None
+            else [str(x) for x in range(len(S))]
+        )
+        self.S = get_formatted_S(S, ts_names)
+        self.tags = {key: np.array(val) for key, val in tags.items()}
+        self.intervals_method = intervals_method
+        self.config = ModelConfig(
+            quantile_levels=quantile_levels,
+            n_jobs=n_jobs,
+        )
+        self.base_reconciliation_model_name = (
+            f"{repr(self.models[0])}/"
+            f"{_build_fn_name(self.hrec.reconcilers[0])}"
+        )
+
+        self.fitted = (
+            repr(self.models[0]) not in models_without_fitted_capability
+        )
+
+    def predict_item(self, entry: DataEntry) -> QuantileForecast:
+        kwargs = {}
+        if (
+            self.config.intervals is not None
+            and "forecast_proportions"
+            not in _build_fn_name(self.hrec.reconcilers[0])
+        ):
+            kwargs["level"] = self.config.intervals
+
+        Y_df = format_data_entry(entry, self.S)
+
+        # set up forecaster
+        forecaster = StatsForecast(
+            df=Y_df,
+            models=self.models,
+            freq=entry["start"].freq,
+            n_jobs=self.config.n_jobs,
+        )
```

Comment:
How does this `StatsForecast` class work? It fits different models independently? And its output is what?

I see in the constructor that `self.models` is instantiated to a singleton model, so I'm wondering whether it is necessary to use this class?

Reply:
This is an interesting point. In short: calling `self.models` is a bit more convolved in this case as we have multiple time series - one per element in the given hierarchy.

When using the class StatsForecast, it automatically handles these multiple time series via the class [GroupedArray](https://github.com/Nixtla/statsforecast/blob/main/statsforecast/core.py#L32). This class is done in a way that it generates more class attributes that do not make it clear how to get out of the logics of StatsForecast. For instance, forecasts, in our case, are computed with [self.ga.forecast](https://github.com/Nixtla/statsforecast/blob/cb80fffc33655f5c7f6cbf44bb3baa4347e431d1/statsforecast/core.py#L750).

With the class GroupedArray, formatting of forecasts is done automatically, and makes it easier for us by avoiding to handle this ourselves. Of course we could explore this further.